### PR TITLE
feat(protocol): implement `Bridge.isMessageFailed`

### PR DESCRIPTION
**Pull request by dantaik**
On the source chain of a message, function `isMessageFailed` can verify if a message has been failed on the destination chain.

A test shall be added later in a separate PR.

### CODE REVIEW

**Review by cyberhorsey — DISMISSED** [review]

File: packages/protocol/test/libs/LibTrieProof.test.ts [hunk 1]
```diff
@@ -62,6 +62,7 @@ describe("integration:LibTrieProof", function () {
             libraries: {
                 LibBridgeProcess: libBridgeProcess.address,
                 LibBridgeRetry: libBridgeRetry.address,
+                LibTrieProof: testLibTreProof.address,
```
Comment (RogerLamTd) [inline packages/protocol/test/libs/LibTrieProof.test.ts:65]: Is this specifically for testing purposes? I tried running this with libTrieProof.address and it worked still

**Review by RogerLamTd — DISMISSED** [review]

File: packages/protocol/test/libs/LibTrieProof.test.ts (hunk 1 above)
Comment (dantaik) [reply packages/protocol/test/libs/LibTrieProof.test.ts:65]: For testing `isMessageFailed`, using testLibTrieProof allows us to modify the destChain's state directly using a new method in TestLibTrieProof.

**Comment by dantaik** [comment]
@davidtaikocha somehow integration & genesis tests are failing. Can you take a look?

**Comment by davidtaikocha** [comment]
> @davidtaikocha somehow integration & genesis tests are failing. Can you take a look?

Opened a PR to fix this: https://github.com/taikoxyz/taiko-mono/pull/13006 , looks like go-ethereum pushed a new `latest` tag image 10 hours ago, but they didn't release a new version of go-ehtereum ( + corresponding change logs )... and that newly pushed `latest` tag image can not be started successfully, so i changed the image tag to the latest version tag `v1.10.26` in that PR for now.

**Review by cyberhorsey — APPROVED** [review]

**Review by davidtaikocha — APPROVED** [review]

**Pull request merged**